main.nf

```diff
@@ -0,0 +1,684 @@
+    file './tmp/fusions.results.total.tsv' into ericscript_fusions
+    file '*' into ericscript_output
+
+    script:
+    """
+    ericscript.pl \\
+        -db ${reference} \\
+        -name fusions \\
+        -p ${task.cpus} \\
+        -o ./tmp \\
+        ${reads[0]} \\
+        ${reads[1]}
+    """
+}
+
+/*
+ * Pizzly
+ */
+process pizzly {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Pizzly", mode: 'copy'
+
+    when:
+    params.pizzly && (!params.singleEnd || params.test)
+
+    input:
+    set val(name), file(reads) from read_files_pizzly
+    file fasta from pizzly_fasta.collect()
+    file gtf from pizzly_gtf.collect()
+    
+    output:
+    file '*.unfiltered.json' into pizzly_fusions
+    file '*.{json,txt,tsv,fasta}' into pizzly_output
+
+    script:
+    """
+    kallisto index -i index.idx -k ${params.pizzly_k} ${fasta}
+    kallisto quant -t ${task.cpus} -i index.idx --fusion -o output ${reads[0]} ${reads[1]}
+    pizzly -k ${params.pizzly_k} \\
+        --gtf ${gtf} \\
+        --cache index.cache.txt \\
+        --align-score 2 \\
+        --insert-size 400 \\
+        --fasta ${fasta} \\
+        --output pizzly_fusions output/fusion.txt
+    """
+}
+
+/*
+ * Squid
+ */
+process squid {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Squid", mode: 'copy'
+
+    when:
+    params.squid && (!params.singleEnd || params.test)
+
+    input:
+    set val(name), file(reads) from read_files_squid
+    file index from star_index.collect()
+    file gtf
+    
+    output:
+    file '*_annotated.txt' into squid_fusions
+    file '*' into squid_output
+
+    script:
+    def avail_mem = task.memory ? "--limitBAMsortRAM ${task.memory.toBytes() - 100000000}" : ''
+    """
+    STAR \\
+        --genomeDir ${index} \\
+        --sjdbGTFfile ${gtf} \\
+        --runThreadN ${task.cpus} \\
+        --readFilesIn ${reads[0]} ${reads[1]} \\
+        --twopassMode Basic \\
+        --chimOutType SeparateSAMold --chimSegmentMin 20 --chimJunctionOverhangMin 12 --alignSJDBoverhangMin 10 --outReadsUnmapped Fastx --outSAMstrandField intronMotif \\
+        --outSAMtype BAM Unsorted ${avail_mem} \\
+        --readFilesCommand zcat
+    samtools sort Aligned.out.bam > Aligned.out.sorted.bam
+    samtools view -Shb Chimeric.out.sam > Chimeric.out.bam
+    squid -b Aligned.out.sorted.bam -c Chimeric.out.bam -o fusions
+    AnnotateSQUIDOutput.py ${gtf} fusions_sv.txt fusions_annotated.txt
+    """
+}
+
+/*************************************************************
+ * Summarizing results from tools
+ ************************************************************/
+process summary {
+    tag "$name"
+    publishDir "${params.outdir}/Report-${name}", mode: 'copy'
+ 
+    when:
+    !params.test && (params.fusioncatcher || params.star_fusion || params.ericscript || params.pizzly || params.squid)
+    
+    input:
+    set val(name), file(reads) from read_files_summary
+    file fusioncatcher from fusioncatcher_fusions.ifEmpty('')
```

Comment:
How does this work when running with multiple samples? How do you know that the output channels are for the same sample? I feel like you need to pass `val(name)` into all of the output channels and then use this to ensure that they correspond when coming in here. Ask @MaxUlysse for help..

Reply:
This looks like 2.0 version. Right now we will only support analysis per sample.

Reply:
@ewels is right about this behavior: We had that issue in nf-core/bcellmagic when not passing the values together with the files as there is no guarantee that the input is sorted in the very same way. Might be worth having this in place for 1.0 already, as its not that much work and will save you tons of it later, but I also understand if you want to have this in V2.0... - but maybe mention this in the docs that you just support single sample analysis or something?

Reply:
I think it needs more than docs - either hardcode the pipeline so that it won't run with more than one sample, or fix this.

It shouldn't be hard to fix, plenty of existing examples already. For example methylseq: [creating channels from files](https://github.com/nf-core/methylseq/blob/1f0eaa6f1a8342e98c87b639553f54c47536f9c3/main.nf#L144-L163), [using channels](https://github.com/nf-core/methylseq/blob/1f0eaa6f1a8342e98c87b639553f54c47536f9c3/main.nf#L346) and [creating new output channels again](https://github.com/nf-core/methylseq/blob/1f0eaa6f1a8342e98c87b639553f54c47536f9c3/main.nf#L350).

Reply:
Ignoring batch mode for now. Testing for one paired-end sample. Can be updated later.